### PR TITLE
feat: fx-impl

### PR DESCRIPTION
**Pull request by geka-evk**
feat: fx-impl

### CODE REVIEW

**Review by kleyow — APPROVED** [review]

File: src/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import http from 'http'
 import { KeyObject } from 'tls'
 import {
-    v1_1 as fspiopAPI,
+    v1_1 as fspiopAPI, // todo: use v2 of openApi
```
Comment (vijayg10) [inline src/index.d.ts:4]: Can we update this to v2?

File: src/lib/ilp/index.js [hunk 2]
```diff
@@ -35,14 +35,7 @@ class Ilp {
...
      */
-    getResponseIlp(transactionObject) {
-        const ilpData = Buffer.from(base64url(JSON.stringify(transactionObject)));
-        const packetInput = {
-            amount: this._getIlpCurrencyAmount(transactionObject.amount), // unsigned 64bit integer as a string
-            account: this._getIlpAddress(transactionObject.payee), // ilp address
-            data: ilpData // base64url encoded attached data
-        };
-
+    getResponseIlp(transactionObject, packetInput) {
```
Comment (vijayg10) [inline src/lib/ilp/index.js:38]: I think this is a breaking change. For example TTK uses this function https://github.com/mojaloop/ml-testing-toolkit/blob/c03910a87a43ca521858c264ef3ca2cb324a19ab/src/lib/mocking/middleware-functions/ilpModel.js#L58
And we don't know if there are any other repos using this. So we need to make this refactoring without breaking anything. Like leave this function like before and if you want, you can add a deprecation comment and have another new function which can serve the new functionality.

File: src/lib/ilp/index.js [hunk 3]
```diff
@@ -68,17 +61,60 @@ class Ilp {
...
+            account: this._getIlpAddress(transactionObject.payee) // ilp address
+        });
+    }
+
+    makeFxQuotePacketInput(transactionObject) {
+        return Object.freeze({
+            data: this.makeIlpData(transactionObject),
+            account: this._getFxIlpAddress(transactionObject.conversionTerms), // ilp address
+            amount: this._getIlpCurrencyAmount(transactionObject.conversionTerms.sourceAmount)
+            // todo: discuss what amount we should use for FX: sourceAmount or targetAmount
```
Comment (vijayg10) [inline src/lib/ilp/index.js:109]: Let's discuss this with @PaulGregoryBaker and remove the comment before merging to main.

File: src/index.d.ts (hunk 1 above)
Comment (geka-evk) [reply src/index.d.ts:4]: for now we can't just use v2 from `api-snippets`, coz there's no appropriacte export on this version:
https://github.com/mojaloop/api-snippets/blob/main/src/index.ts
So we need first to publish a new release of `api-snippets`

File: src/lib/ilp/index.js (hunk 2 above)
Comment (geka-evk) [reply src/lib/ilp/index.js:38]: yes, you're right.
I've updated impl. and added unit-test for this method

File: src/lib/ilp/index.js (hunk 3 above)
Comment (geka-evk) [reply src/lib/ilp/index.js]: Use `'0'` string as amount for fxQuotes, coz empty string causes an error:
image

**Review by vijayg10 — APPROVED** [review]

File: src/index.d.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import http from 'http'
 import { KeyObject } from 'tls'
 import {
-    v1_1 as fspiopAPI,
+    v2_0 as fspiopAPI,
```
Comment (elnyry-sam-k) [inline src/index.d.ts:4]:
```suggestion
    //This needs reconsidering if and when more changes are included in fspiop v2.0, currently they're non-breaking as far as any existing fields are replaced, but when such changes come in, this needs to be reviewed.
    v2_0 as fspiopAPI,
```

**Comment by elnyry-sam-k** [comment]
Thanks @geka-evk , one caveat to be added as comment but otherwise, looks good, thanks !

**Review by elnyry-sam-k — DISMISSED** [review]

**Review by elnyry-sam-k — APPROVED** [review]

**Pull request merged**